### PR TITLE
[Convert method reference]Lps 163778

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-163778

/cc @tinatian @dantewang

### CODE REVIEW

**Comment by slnn** [comment]
Hi @zhangyan0701 

Could you help to scan test results when they are finished?

Thanks
Lily

File: modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java [hunk 1]
```diff
@@ -402,6 +403,11 @@ public void updateApps(
 		writeJSON(actionRequest, actionResponse, jsonObject);
 	}
 
+	@Activate
+	protected void activate() {
+		super.oAuthManager = _oAuthManager;
```
Comment (dantewang) [inline modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java:408]: @slnn Did you try to directly add `@Reference` to the protected field in the super class? It should work even if it's not a component?

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java (hunk 1 above)
Comment (slnn) [reply modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java:408]: Hi @dantewang 

We can not do that, SF will complains:

1: @Reference should not be used in a class without @Component, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/java_module_component_check.markdown: /home/liferay/repo/portal-ce/modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/RemoteMVCPortlet.java (SourceCheck:JavaModuleComponentCheck)

Comment (slnn) [reply modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java:408]: The supper class RemoteMVCPortlet is not an abstract class.

**Comment by zhangyan0701** [comment]
Some errors occur during portal startup:
```
[beanshell] 2022-09-23 04:57:14.158 ERROR [Portal Dependency Manager Component Executor--1][MarketplacePurchasedPortlet:56] bundle com.liferay.marketplace.store.web:7.0.27 (798)[com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet(7776)] : findField: Suitable but non-accessible field _appLocalService in class com.liferay.marketplace.store.web.internal.portlet.MarketplaceStorePortlet, subclass of component class com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet
[beanshell] 2022-09-23 04:57:14.160 ERROR [Portal Dependency Manager Component Executor--1][MarketplacePurchasedPortlet:56] bundle com.liferay.marketplace.store.web:7.0.27 (798)[com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet(7776)] : findField: Suitable but non-accessible field _appService in class com.liferay.marketplace.store.web.internal.portlet.MarketplaceStorePortlet, subclass of component class com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet
[beanshell] 2022-09-23 04:57:14.160 ERROR [Portal Dependency Manager Component Executor--1][MarketplacePurchasedPortlet:56] bundle com.liferay.marketplace.store.web:7.0.27 (798)[com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet(7776)] : findField: Suitable but non-accessible field _oAuthManager in class com.liferay.marketplace.store.web.internal.portlet.MarketplaceStorePortlet, subclass of component class com.liferay.marketplace.store.web.internal.portlet.MarketplacePurchasedPortlet
```

Full console log:
https://testray.liferay.com/reports/production/logs/2022-09/test-1-38/test-portal-acceptance-pullrequest(master)/3872/functional-smoke-tomcat90-mysql57-jdk11_open/0/0/jenkins-console.txt.gz?authuser=0

**Comment by slnn** [comment]
ci:test:relevant

**Comment by slnn** [comment]
Hi @zhangyan0701 

I fix the issue and forced push to this branch, please help to rescan test results when they are finished.

Thanks
Lily

**Comment by catarineaguiar** [comment]
no regressions.

**Comment by slnn** [comment]
ci:test:relevant

**Comment by slnn** [comment]
ci:test:sf

File: modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java (hunk 1 above)
Comment (slnn) [reply modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java]: Hi @dantewang 

I have updated this pull to change the super class to an abstract class and then add abstract get method in it in order to make subclass to implement it.

Lily

**Comment by slnn** [comment]
Hi @zhangyan0701 

Could you help to rescan the test results when they are finished?

Thanks
Lily

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java [hunk 2]
```diff
@@ -500,18 +505,15 @@ protected void processPortletParameterMap(
...
-	}
-
 	@Reference
 	protected AppLocalService appLocalService;
 
 	@Reference
 	protected AppService appService;
 
+	@Reference
+	protected OAuthManager oAuthManager;
```
Comment (tinatian) [inline modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java:515]: @slnn Please make this change ins separate commit to highlight the reason you need a "protected" here

**Comment by tinatian** [comment]
@slnn This commit fixes a bug already existed, right? https://github.com/liferay-core-infra/liferay-portal/pull/1790/commits/8f6d8e3fc145b13fbab690d1a901e82e2dfdac5f.

If yes, please create a ticket to fix this bug and send it first,then resume this task.

File: modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java (hunk 2 above)
Comment (slnn) [reply modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java:515]: I have separate the commit to highlight the reason that needs a "protected"  in the new update

**Comment by slnn** [comment]
Hi @tinatian 

I update this pull with updating more details in commit message to explain why I change these private fields to protected fields.


For this, it is a bug after I convert these method references to field references, otherwise, I do not reproduce it.

So I just update the commit message with the error to explain why I change the field to protected.

> @slnn This commit fixes a bug already existed, right? [8f6d8e3](https://github.com/liferay-core-infra/liferay-portal/commit/8f6d8e3fc145b13fbab690d1a901e82e2dfdac5f).
> 
> If yes, please create a ticket to fix this bug and send it first,then resume this task.

**Comment by tinatian** [comment]
@slnn Please merge the first two commits to show people that this is not an existing bug, this is required as you removed a protected set method which needs by a sub class.

Tina
